### PR TITLE
todo view and source navigation

### PR DESCRIPTION
**Pull request by kr-g**
this adds a new view in south panel where all todo's from the source are extracted and shown in a list.
when clicking on a todo (list item) the corresponsing line in editor code view is selected.

### CODE REVIEW

**Comment by aivarannamaa** [comment]
This is good goal! Not really a beginner's thing, but as it's not prominent it's OK. 

Please keep different features separated in PR-s!

**Comment by aivarannamaa** [comment]
In case the file doesn't have any TODO-s, I would show a single item, which describes the purpose of this view.

**Comment by kr-g** [comment]
Feel free to change !
For me it is already fine in the current state.

**Comment by kr-g** [comment]
> In case the file doesn't have any TODO-s, I would show a single item, which describes the purpose of this view.

as supposed by you @aivarannamaa

**Comment by kr-g** [comment]
cleaned PR from unrelated source

RFC

**Review by aivarannamaa — CHANGES_REQUESTED** [review]

File: thonny/plugins/todo_view.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        if iid != "":
+            values = self.tree.item(iid)["values"]
+            line_no = values[0]
+            if line_no:
+                editor = get_workbench().get_editor_notebook().get_current_editor()
+                editor.select_line(line_no)
+
+
+def load_plugin() -> None:
+    get_workbench().add_view(TodoView, tr("Todo's"), "s")
```
[the remark this answers is not in the document]
Comment (kr-g) [reply thonny/plugins/todo_view.py:105]: i m also not 100% sure regarding this.
so i checked that (german version)

https://english.lingolia.com/de/schreibschule/zeichensetzung/apostroph
https://dict.leo.org/grammatik/englisch/nouns_plural_context.php?lang=de#id=2.1.4.d

well... could be right or not.

how about just "TODO" ?

File: thonny/plugins/todo_view.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+        editor = get_workbench().get_editor_notebook().get_current_editor()
+
+        if editor is None:
+            self._current_code_view = None
+            self._current_source = None
+            return
+
+        new_cw = editor.get_code_view()
+        new_source = new_cw.get_content_as_bytes()
```
[the remark this answers is not in the document]
Comment (kr-g) [reply thonny/plugins/todo_view.py:62]: i made that PR half year ago. 
i have no idea why i made that decision.

File: thonny/plugins/todo_view.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+        self.tree["show"] = ["headings"]
+
+        self._current_code_view = None
+        self._current_source = None
+
+        self._update(None)
+
+    def _update(self, event):
+
+        # todo: not updated on first run
```
[the remark this answers is not in the document]
Comment (kr-g) [reply thonny/plugins/todo_view.py]: done

File: thonny/plugins/todo_view.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kr-g) [reply thonny/plugins/todo_view.py]: removed

File: thonny/plugins/todo_view.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        self.tree.bind("<<TreeviewSelect>>", self._on_click)
+        self.tree.bind("<Map>", self._update, True)
+
+        # todo bind to reload event for update
+
+        get_workbench().get_editor_notebook().bind("<<NotebookTabChanged>>", self._update)
+        get_workbench().bind("Save", self._update, True)
+        get_workbench().bind("SaveAs", self._update, True)
+        get_workbench().bind_class("Text", "<Double-Button-1>", self._update, True)
+
```
[the remark this answers is not in the document]
Comment (kr-g) [reply thonny/plugins/todo_view.py:38]: done

File: thonny/plugins/todo_view.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kr-g) [reply thonny/plugins/todo_view.py]: change to TODO

**Comment by kr-g** [comment]
ok thanks for the feedback.
learned a bit more about events.

RFC

**Comment by aivarannamaa** [comment]
Thank you!

**Pull request merged**